test/unit/controllers/VersionsControllerSpec.scala

```diff
@@ -0,0 +1,294 @@
+    protected val stubValidateAndExtractHeadersAction: ValidateAndExtractHeadersAction = new ValidateAndExtractHeadersAction(new HeaderValidator(mockInformationLogger))
+    protected val stubVersionResponseFilterService: VersionResponseFilterService = new VersionResponseFilterService()
+    protected val stubDeclarationVersionsService = new DeclarationVersionService(stubVersionResponseFilterService, mockApiSubscriptionFieldsConnector, mockInformationLogger, mockVersionConnector, mockDateTimeService, stubUniqueIdsService)
+    protected val stubConversationIdAction = new ConversationIdAction(stubUniqueIdsService, mockInformationLogger)
+
+    protected val controller: VersionsController = new VersionsController(
+      stubShutterCheckAction,
+      stubValidateAndExtractHeadersAction,
+      stubAuthStatusAction,
+      stubConversationIdAction,
+      stubInternalClientIdsCheckAction,
+      stubDeclarationVersionsService,
+      Helpers.stubControllerComponents(),
+      mockInformationLogger) {}
+
+    protected def awaitSubmitMrn(request: Request[AnyContent], declarationSubmissionChannel: Option[String] = None): Result = {
+      controller.list(mrnValue, declarationSubmissionChannel).apply(request)
+    }
+
+    protected def submitMrn(request: Request[AnyContent], declarationSubmissionChannel: Option[String] = None): Future[Result] = {
+      controller.list(mrnValue, declarationSubmissionChannel).apply(request)
+    }
+
+    when(mockVersionConnector.send(any[DateTime], meq[UUID](correlationId.uuid).asInstanceOf[CorrelationId], any[ApiVersion], any[Option[ApiSubscriptionFieldsResponse]], meq[SearchType](mrn))(any[AuthorisedRequest[_]])).thenReturn(Future.successful(stubHttpResponse))
+    when(mockDateTimeService.nowUtc()).thenReturn(dateTime)
+    when(mockApiSubscriptionFieldsConnector.getSubscriptionFields(any[ApiSubscriptionKey])(any[ValidatedHeadersRequest[_]], any[HeaderCarrier])).thenReturn(Future.successful(apiSubscriptionFieldsResponse))
+  }
+
+  private val errorResultBadgeIdentifier = errorBadRequest("X-Badge-Identifier header is missing or invalid").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+  private val errorResultMissingIdentifiers = errorBadRequest("Both X-Submitter-Identifier and X-Badge-Identifier headers are missing").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+
+  private val errorResultMrnTooLong = errorBadRequest("MRN too long").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+  private val errorResultMissingMrn = errorBadRequest("Missing MRN").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+  private val errorResultCDS60002 = errorBadRequest("MRN invalid", "CDS60002").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
+
+  "Declaration Version Controller for MRN queries" should {
+    "process CSP request when call is authorised for CSP" in new SetUp() {
+      authoriseCsp()
+
+      awaitSubmitMrn(ValidCspDeclarationStatusRequest)
```

Comment:
If `ValidCspDeclarationStatusRequest` works for both status and version endpoints then give it a more neutral name e.g. `ValidCspDeclarationRequest`